### PR TITLE
35099 Pagination migration ESLint rule

### PR DESCRIPTION
**Pull request by andresriveratoro**
## Description
This PR adds an ESLint warning rule to prompt developers to migrate from using the component-library's React `Pagination` component to the new `va-pagination` web component.

IMPORTANT: `onPageSelect` changed to be an event and developers will need to update `onPageSelect` handlers to retrieve the page from `event.detail.page`. This script doesn't handle the updating of onPageSelect handlers.

## Original issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/35099

## Testing done
Screen Shot 2022-03-09 at 14 54 21

## Screenshots
Screen Shot 2022-03-09 at 15 00 24

## Acceptance criteria
- [x] Create ESLint rule to ease migration from React `Pagination` component to `va-pagination` (VaPagination)
- [x] Create tests for ESLint rule

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by pdavies88 — APPROVED** [review]

**Review by timwright12 — APPROVED** [review]
Feel free to update codeowners for the linting files you all own

**Comment by k80bowman** [comment]
@timwright12 I opened a PR for updating code ownership https://github.com/department-of-veterans-affairs/vets-website/pull/20468

**Comment by andresriveratoro** [comment]
I'm not sure why these extra files show as changed here

File: src/applications/gi/containers/CompareLayout.jsx [hunk 1]
```diff
@@ -373,12 +373,9 @@ const CompareLayout = ({
...
                   <div className="caution-flag-icon vads-u-flex--1">
                     {!hasFlags && (
-                      <i
-                        className={`fa fa-check`}
-                        style={{ display: 'none' }}
-                      />
+                      <i className="fa fa-check" style={{ display: 'none' }} />
                     )}
-                    {hasFlags && <i className={`fa fa-exclamation-triangle`} />}
+                    {hasFlags && <i className="fa fa-exclamation-triangle" />}
```
Comment (timwright12) [inline src/applications/gi/containers/CompareLayout.jsx:378]:
```suggestion
                    {hasFlags && <i className="fa fa-exclamation-triangle" aria-hidden="true" />}
```

**Review by timwright12 — APPROVED** [review]

**Pull request closed without merging**